src/MultiAgentReinforcementLearning/src/MultiAgentReinforcementLearning.jl

```diff
@@ -0,0 +1,224 @@
+module MultiAgentReinforcementLearning
+
+export MultiAgentPolicy
+export MultiAgentHook
+
+using ReinforcementLearningBase
+using ReinforcementLearningCore
+
+import ReinforcementLearningCore: RLCore
+import Base.getindex
+import Base.iterate
+
+struct MultiAgentPolicy <: AbstractPolicy
+    agents::NamedTuple
+
+    function MultiAgentPolicy(agents::NamedTuple)
+        new(agents)
+    end
+end
+
+struct MultiAgentHook <: AbstractHook
+    hooks::NamedTuple
+
+    function MultiAgentHook(hooks::NamedTuple)
+        new(hooks)
+    end
+end
+
+
+(p::MultiAgentPolicy)(env::AbstractEnv) = nothing # Default does nothing, but might be useful for some environments to clean up / pass final state to agents
+
+struct CurrentPlayerIterator
+    env::AbstractEnv
+end
+
+function current_player_iterator(env)
+    return CurrentPlayerIterator(env)
+end
+
+Base.iterate(current_player_iterator::CurrentPlayerIterator) =
+    (current_player(current_player_iterator.env), current_player_iterator.env)
+
+Base.iterate(current_player_iterator::CurrentPlayerIterator, env) =
+    (current_player(current_player_iterator.env), current_player_iterator.env)
```

Comment:
These seem to do the same, env is not needed. We can simplify to 
```suggestion

Base.iterate(current_player_iterator::CurrentPlayerIterator, env = current_player_iterator.env) =
    (current_player(current_player_iterator.env), env)
```
would allow a specific env argument, otherwise behaves as the two above functions.